include/boost/numeric/ublas/tensor/functions.hpp

```diff
@@ -151,21 +124,50 @@ namespace boost::numeric::ublas
                 "error in boost::numeric::ublas::prod(ttv): first "
                 "argument tensor should not be empty.");
 
-        if (b.size() == 0)
+        if (b.empty())
             throw std::length_error(
                 "error in boost::numeric::ublas::prod(ttv): second "
                 "argument vector should not be empty.");
 
-        auto nc = detail::extents_result_tensor_times_vector(a.extents());
-        auto nb = std::vector<typename extents_type::value_type>{b.size(), 1};
+        using extents_value_type = typename extents_type::value_type;
 
         auto a_extents = a.extents();
+
+        auto extents_result = [&a_extents](){
+            using size_type = typename extents_type::size_type;
+            if constexpr( is_static_rank_v<extents_type> ){
+                // To disable warning for unused variable;
+                (void)a_extents;
+                constexpr size_type esz = extents_type::_size - 1u;
+                constexpr auto sz = std::max( esz, size_type(2) );
+                auto ret = extents< sz >();
+                ret.fill(1u);
+                return ret;
+            }else{
+                using extents_base_type = typename extents_type::base_type;
+                auto const sz = std::max( a_extents.size() - 1, size_type(2) );
+                auto arr = extents_base_type(sz,1);
+                return extents_type{ std::move(arr) } ;
+            }
+        };
```

Comment:
Just a minor comment:

```cpp
if constexpr (is_static_rank_v<extents_type>) {
  constexpr auto size = std::max(a_extents.size()-1u,2u);  // a_extents.size() should be constexpr
  auto shape = extents<size>(); // we should make e.g. extents<size>(1) or extents<size>{1}
  shape.fill(1);
  return shape;
}
else{
  auto size = std::max(a_extents.size()-1u,2u);  // a_extents.size() 
  auto shape = extents(typename extents_type::base_type(size));
  shape.fill(1u);
  return shape; 
}

Reply:
But just leave it like it is. We can do several PRs based on your changes.

Reply:
1. The problem with `extents<size>(1)` is that it accepts `std::initializer_list` which will hinder or try to match with it. That's why I added the function `fill`.

2. I avoided `a_extents.size()` because of msvc's bugs which I faced last time I used it. I don't know if msvc has improved it's `constexpr` capabilities.